paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -87,6 +87,19 @@ public interface Catalog extends AutoCloseable {
     void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
             throws DatabaseNotExistException, DatabaseNotEmptyException;
 
+    /**
+     * Rename a database.
```

Comment:
Adds the blew note for renaming database:

```
     * <p>NOTE: If you use object storage, such as S3 or OSS, please use this syntax carefully,
     * because the renaming of object storage is not atomic, and only partial files may be moved in
     * case of failure.
```

Reply:
ok,I will do it.